bundled/linter/utils.py

```diff
@@ -184,7 +169,19 @@ def run_module(
     return LinterResult(str_output.get_value(), str_error.get_value())
 
 
-def run_path(argv: Sequence[str], use_stdin: bool, source: str = None) -> LinterResult:
+def run_module(
+    module: str, argv: Sequence[str], use_stdin: bool, cwd: str, source: str = None
+) -> LinterResult:
+    """Runs linter as a module."""
+    if is_same_path(os.getcwd(), cwd):
+        return _run_module(module, argv, use_stdin, source)
+    with change_cwd(cwd):
```

Comment:
Shouldn't the lock cover the above _run_module too? Otherwise it might get changed while it's running. I guess the use of a condition variable would allow jobs to run in parallel if the have the same desired CWD.

Reply:
You are correct. The lock should be held in both cases.